common/quotas/rate_limiter_impl.go

```diff
@@ -141,7 +135,8 @@ func (rl *RateLimiterImpl) refreshInternalRateLimiterImpl(
 	}
 
 	if refresh {
-		limiter := rate.NewLimiter(rate.Limit(rl.rate), rl.burst)
-		rl.goRateLimiter.Store(limiter)
+		now := time.Now()
+		rl.goRateLimiter.SetLimitAt(now, rate.Limit(rl.rate))
```

Comment:
i assume these are thread-safe operations then and we don't have to worry about processor caching stale values or anything like that?

Reply:
it is thread safe: 
https://github.com/golang/time/blob/f8bda1e9f3ba/rate/rate.go#L280